Pass newRepositoryData to cleanupUnlinkedRootAndIndicesBlobs

DaveCTurner opened the pull request:
During cleanup these `RepositoryData` objects refer to the same
snapshots so it doesn't really matter, but for the sake of consistency
we should use the updated `RepositoryData` here.

Relates a comment on #100657

elasticsearchmachine commented:
Pinging @elastic/es-distributed (Team:Distributed)

DaveCTurner commented:
Ha ok this inconsistency turns out to be enforced by tests, although I'm still pretty sure the change is correct. At the end of `org.elasticsearch.repositories.AbstractThirdPartyRepositoryTestCase#testCleanup` we assert that we _only_ deleted the dangling index, but in fact we should also have cleaned up the stale `index-N` blob. I'm going to leave this alone for now, it's not in the way, but we should definitely fix this behaviour and these tests in future.

ywangd commented:
I traced on past changes and found this comment to be relevant

https://github.com/elastic/elasticsearch/pull/42723/files#r289432902

In summary, I think there is no longer a strong reason to keep the old `index-N` file around.

DaveCTurner committed:
Pass newRepositoryData to cleanupUnlinkedRootAndIndicesBlobs

During cleanup these `RepositoryData` objects refer to the same
snapshots so it doesn't really matter, but for the sake of consistency
we should use the updated `RepositoryData` here.

Relates a comment on #100657

DaveCTurner committed:
Fix up assertCleanupResponse
